"Cannot extend a class 'undefined'." when class is declared in a namespace

Hi,
I noticed the error message when trying to extend a class with a private constructor says `'undefined'` if the class is in a namespace:

**TypeScript Version:**  master (14e5de3b1d6eafa2d3213a72b8599ecc972ed419)

**Code**

```ts
declare namespace abc {
    class XYZ {
        private constructor();
    }
}

class C extends abc.XYZ {
}

```

**Expected behavior:**

```
error TS2675: Cannot extend a class 'abc.XYZ'. Class constructor is marked as private.
```

**Actual behavior:** 
```
error TS2675: Cannot extend a class 'undefined'. Class constructor is marked as private.
```

Thanks!

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -17189,7 +17189,7 @@ namespace ts {
                 if (declaration && getModifierFlags(declaration) & ModifierFlags.Private) {
                     const typeClassDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(type.symbol);
                     if (!isNodeWithinClass(node, typeClassDeclaration)) {
-                        error(node, Diagnostics.Cannot_extend_a_class_0_Class_constructor_is_marked_as_private, (<Identifier>node.expression).text);
+                        error(node, Diagnostics.Cannot_extend_a_class_0_Class_constructor_is_marked_as_private, getFullyQualifiedName(type.symbol));
                     }
                 }
             }

--- a/tests/baselines/reference/extendPrivateConstructorClass.errors.txt
+++ b/tests/baselines/reference/extendPrivateConstructorClass.errors.txt
@@ -1,0 +1,15 @@
+tests/cases/compiler/extendPrivateConstructorClass.ts(7,17): error TS2675: Cannot extend a class 'abc.XYZ'. Class constructor is marked as private.
+
+
+==== tests/cases/compiler/extendPrivateConstructorClass.ts (1 errors) ====
+    declare namespace abc {
+        class XYZ {
+            private constructor();
+        }
+    }
+    
+    class C extends abc.XYZ {
+                    ~~~~~~~
+!!! error TS2675: Cannot extend a class 'abc.XYZ'. Class constructor is marked as private.
+    }
+    

--- a/tests/baselines/reference/extendPrivateConstructorClass.js
+++ b/tests/baselines/reference/extendPrivateConstructorClass.js
@@ -1,0 +1,24 @@
+//// [extendPrivateConstructorClass.ts]
+declare namespace abc {
+    class XYZ {
+        private constructor();
+    }
+}
+
+class C extends abc.XYZ {
+}
+
+
+//// [extendPrivateConstructorClass.js]
+var __extends = (this && this.__extends) || function (d, b) {
+    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+    function __() { this.constructor = d; }
+    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
+};
+var C = (function (_super) {
+    __extends(C, _super);
+    function C() {
+        return _super.apply(this, arguments) || this;
+    }
+    return C;
+}(abc.XYZ));

--- a/tests/cases/compiler/extendPrivateConstructorClass.ts
+++ b/tests/cases/compiler/extendPrivateConstructorClass.ts
@@ -1,0 +1,8 @@
+declare namespace abc {
+    class XYZ {
+        private constructor();
+    }
+}
+
+class C extends abc.XYZ {
+}

